pointcloud/basic_octree.go

```diff
@@ -132,9 +140,103 @@ func (octree *BasicOctree) MetaData() MetaData {
 	return octree.meta
 }
 
+// Pose returns the pose of the octree.
+func (octree *BasicOctree) Pose() spatialmath.Pose {
+	return spatialmath.NewPoseFromPoint(octree.center)
+}
+
+// AlmostEqual compares the octree with another geometry and checks if they are equivalent.
+func (octree *BasicOctree) AlmostEqual(geom spatialmath.Geometry) bool {
+	// TODO
+	return false
+}
+
+// Transform recursively steps through the octree and transforms it by the given pose.
+func (octree *BasicOctree) Transform(p spatialmath.Pose) spatialmath.Geometry {
```

Comment:
This needs to either be properly written, using a `Pose` for the outermost octree, or else it must return `nil` and thus cause a failure for anything trying to use a transformed octree.

What we have now is something that looks similar to what a working implementation would look like, but will actively place the octree in the wrong location, and thus will prevent the collision avoidance system from avoiding collisions.

Reply:
it's now returning nil